src/forms/text.rs

```diff
@@ -0,0 +1,387 @@
+                match code {
+                    KeyCode::Backspace => self.remove_char(1),
+                    KeyCode::Delete => self.remove_char(-1),
+                    KeyCode::Left => self.move_cursor(-1),
+                    KeyCode::Right => self.move_cursor(1),
+                    KeyCode::Home => self.move_cursor(i32::MIN),
+                    KeyCode::End => self.move_cursor(i32::MAX),
+                    KeyCode::Char(c) => {
+                        if modifiers.is_empty() {
+                            self.put_char(*c);
+                        }
+                        if *modifiers == KeyModifiers::SHIFT {
+                            // I don't understand why it works this way but not the other
+                            if c.is_ascii_uppercase() {
+                                self.put_char(c.to_ascii_uppercase());
+                            } else {
+                                self.put_char(c.to_ascii_lowercase());
+                            }
+                        }
+                    }
+                    _ => {}
+                }
+            }
+        }
+    }
+
+    fn set_active(&mut self, active: bool) {
+        self.active = active;
+    }
+
+    fn is_active(&self) -> bool {
+        self.active
+    }
+
+    fn validate(&self, validation_result: &mut FormValidationResult) {
+        self.self_validate(validation_result);
+    }
+
+    fn get_output(&self) -> FormValue {
+        FormValue::String(self.input_buffer.to_string())
+    }
+
+    fn set_options(&mut self, options: FormOptions) {
+        self.options = options
+    }
+
+    fn get_options(&self) -> &FormOptions {
+        &self.options
+    }
+
+    fn draw(&mut self, tick: usize) -> &Screen {
+        if self.dirty {
+            self.screen.clear();
+            self.screen.print_fbg(
+                if self.cursor_pos >= self.screen.get_width() as usize {
+                    -((self.cursor_pos - self.screen.get_width() as usize) as i32) - 1
+                } else {
+                    0
+                },
+                0,
+                &self.input_buffer,
+                self.options.style.fg,
+                self.options.style.bg,
+            );
+            self.dirty = false;
+        }
+        let current_cursor_pos =
+            std::cmp::min(self.cursor_pos as i32, self.screen.get_width() as i32 - 1);
+        if self.active && tick % 2 == 0 {
+            if let Ok(mut cursor_pxl) = self.screen.get_pxl(current_cursor_pos, 0) {
+                cursor_pxl.bg = self.options.style.fg;
+                cursor_pxl.fg = self.options.style.bg;
+                self.screen.set_pxl(current_cursor_pos, 0, cursor_pxl);
+            }
+        } else if let Ok(mut cursor_pxl) = self.screen.get_pxl(current_cursor_pos, 0) {
+            cursor_pxl.bg = self.options.style.bg;
+            cursor_pxl.fg = self.options.style.fg;
+            self.screen.set_pxl(current_cursor_pos, 0, cursor_pxl);
+        }
```

Comment:
```rust
if let Ok(mut cursor_pxl) = self.screen.get_pxl(current_cursor_pos, 0) {
    if self.active && tick % 2 == 0 {
        cursor_pxl.bg = self.options.style.fg;
        cursor_pxl.fg = self.options.style.bg;
    }
    else {
        cursor_pxl.bg = self.options.style.bg;
        cursor_pxl.fg = self.options.style.fg;
    }
    self.screen.set_pxl(current_cursor_pos, 0, cursor_pxl);
}
```
I think this would be more readable.